pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/CompareObjectsWithEqualsRule.java

```diff
@@ -84,7 +84,9 @@ public Object visit(ASTEqualityExpression node, Object data) {
                 ASTReferenceType type1 = ((Node) nd1.getAccessNodeParent())
                         .getFirstDescendantOfType(ASTReferenceType.class);
                 // skip, if it is an enum
-                if (type0.getType() != null && type0.getType().equals(type1.getType()) && type0.getType().isEnum()) {
+                if (type0.getType() != null && type0.getType().equals(type1.getType())
+                    // It may be a custom enum class or an explicit Enum class usage
+                    && (type0.getType().isEnum() || type0.getType() == java.lang.Enum.class)) {
```

Comment:
it seems to me, the issue lies on `isEnum()` itself, returning false for something that is an enum. I'd rather change it there than here.

Reply:
You are right about that the problem lies on `isEnum`. However, I've try to do so, but I realized that `isEnum` is a `Class` class method, and that's why I had to change it here.